pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -559,6 +588,147 @@ private RebalanceResult doRebalance(TableConfig tableConfig, RebalanceConfig reb
     }
   }
 
+  private long calculateTableSizePerReplicaInBytes(String tableNameWithType) {
+    long tableSizePerReplicaInBytes = -1;
+    if (_tableSizeReader == null) {
+      LOGGER.warn("tableSizeReader is null, cannot calculate table size for table {}!", tableNameWithType);
+      return tableSizePerReplicaInBytes;
+    }
+    try {
+      TableSizeReader.TableSubTypeSizeDetails tableSizeDetails =
+          _tableSizeReader.getTableSubtypeSize(tableNameWithType, 30_000);
```

Comment:
perhaps add LOGGER.info() before/after this method, to know how much time controller may spend getting the table size, as I assume it'd take a while for large tables

and maybe leave a TODO to make 30s configurable later

Reply:
done